pages/sitemap.xml.ts

```diff
@@ -1,56 +1,115 @@
-import React from 'react';
-import globby from 'globby';
 import { getPrezlyApi } from '@/utils/prezly';
-import { Story } from '@prezly/sdk/dist/types';
-import { NextPageContext } from 'next';
-
-const createSitemap = (url: string, paths: Array<string>, stories: Array<Story>) => `<?xml version="1.0" encoding="UTF-8"?>
-    <urlset xmlns="http://www.sitemaps.org/schemas/sitemap/0.9">
-        ${paths
-        .map((p) => p.replace(/^pages\//, '')) // get rid of /pages
-        .map((p) => p.replace(/\.(tsx|ts)$/, '')) // get rid of file extension
-        .map((path) => `
-                <url>
-                    <loc>${`${url}/${path === 'index' ? '' : path}`}</loc>
-                </url>
-            `)
-        .join('')}
-        ${stories
-        .map(({ slug }) => `
-                <url>
-                    <loc>${`${url}/${slug}`}</loc>
-                </url>
-            `)
-        .join('')}
-    </urlset>
-`;
-
-class Sitemap extends React.Component {
-    static async getInitialProps(ctx: NextPageContext) {
-        const { res, req } = ctx;
-
-        if (!req || !res) { // client side
-            return null;
+import { Category, Story } from '@prezly/sdk/dist/types';
+import { NextPage, NextPageContext } from 'next';
+
+type Url = {
+    loc: string;
+    changefreq?: string;
+    priority?: string;
+};
+
+class SitemapBuilder {
+    private baseUrl: string;
+
+    private urls: Url[] = [];
+
+    constructor(baseUrl: string) {
+        this.baseUrl = baseUrl;
+    }
+
+    addUrl(loc: string) {
+        this.urls.push({
+            loc: this.baseUrl + loc,
+            changefreq: SitemapBuilder.guessFrequency(loc),
+            priority: SitemapBuilder.guessPriority(loc),
+        });
+    }
+
+    private static guessFrequency(loc: string) {
+        if (loc === '/') return 'daily';
+
+        return 'weekly';
+    }
+
+    private static guessPriority(loc: string) {
+        if (loc === '/') return '0.9';
+
+        if (loc.startsWith('/category/')) {
+            return '0.8';
         }
 
-        const paths = await globby([
-            'pages/**/*{.tsx,.ts}',
-            '!pages/**/[*', // ignore dynamic paths
-            '!pages/_*.js', // ignore nextjs special paths
-            '!pages/api', // ignore api routes
-        ]);
+        return '0.7';
+    }
```

Comment:
I implemented the sitemap builder in this PR (alongside all your requests on #5), please review those 🙏🏻 

Also, I put those values for `priority` & `frequency` but I can change them